### ISSUE TITLE
404 - GET http://SERVER_IP_ADRESS:8080/get-basic-data?page=1

### ISSUE THREAD

**Issue by jjdblast** [iam-abbas/Reddit-Stock-Trends#72]
![image](https://user-images.githubusercontent.com/3394810/113690884-7690e900-96fe-11eb-862d-eff3499ed31d.png)

I'm trying to deploy with docker, but when I visit the site by 8080 port, it shows nothing, chrome debug message as above.

I configured `praw.ini` just like below:

```
[ClientSecrets]
client_id=a_14_character_string_from_reddit
client_secret=a_27_character_string_from_reddit
user_agent=my_app_name
```

The backend and frontend container built well and running healthy,  docker logs show no error message. 

I found after I deploy the service by `docker-compose up -d`, there is two new files generated under `back/data` folder: `2021-04-06_financial_df.csv` and  `2021-04-06_tick_df.csv`.

**Comment by iam-abbas** [comment]
You need to run the backend along with the backend

**Comment by jjdblast** [comment]
> You need to run the backend along with the frontend

Hi abbas, thanks for repply, I checked the running container, and pretty sure that backend running well along with the frontend.

`> docker ps | grep reddit-stock` shows:

![image](https://user-images.githubusercontent.com/3394810/113699003-9c6ebb80-9707-11eb-9001-925c50e3d2d7.png)

**Comment by iam-abbas** [comment]
I just checked, it is an issue with the frontend probably. The Dockerfile seems to be the problem.

**Comment by jjdblast** [comment]
I think the problem maybe caused by frontend query the wrong backend port. 

The frontend try to GET data from `:8080/get-basic-data`, but the backend service is actually running on `5006`, I tried query `:5006/get-basic-data` and it returns the right data.

**Comment by jjdblast** [comment]
Also, another dummies question related to this issue: 

It's seems like by current `docker-compose.yml` setup, I can only use [virtual IP or hostname 'backend'](https://docs.docker.com/compose/networking/) to retrive data from backend container. I entered into the frontend container, use `curl localhost:5006/get-basic-data` and get nothing since it's refused.

I'm not famaliar with Vue or React, just checked your frontend code, and don't know where did you set the backend IP or hostname with service port, could you send me to there?

**Comment by iam-abbas** [comment]
https://github.com/iam-abbas/Reddit-Stock-Trends/blob/c9b3b0a4131daf5d00e65db889ce9a1280d49625/front/src/api/ticksData.ts#L4
And
https://github.com/iam-abbas/Reddit-Stock-Trends/blob/c9b3b0a4131daf5d00e65db889ce9a1280d49625/front/.env.example#L1

From what I understand your backend is failing. Can you check the logs? While building docker (with `––no–cache`) and logs with `docker logs –f <backend_container_id>`

**Comment by Tyler-Barham** [comment]
@jjdblast  The Vue frontend is using 8080 for API calls as it defaults to the same address/port as its own server. You need to create a new file `./front/.env` and copy the contents of `./front/.env.example` to it. The `VUE_APP_API_URL` variable within `.env` tells the frontend what the basepath of the API is.

**Comment by jjdblast** [comment]
@Tyler-Barham @iam-abbas 

Thanks for your help, the backend container works well and no error message at all. After some struggling, I created a new `./front/.env` file like :

```
VUE_APP_API_URL=http://<MY_SERVER_EXTERNAL_IP>:5006/
```

and suddenly everything work like a charm.

I'm still a little confused, since when I first time configure the `./front/.env` by `localhost`, it still not worked and error message attached.

```
VUE_APP_API_URL=http://localhost:5006/
```

![image](https://user-images.githubusercontent.com/3394810/113806454-e356c280-9794-11eb-9fb8-fc8663ce9743.png)

I found [a similar question](https://stackoverflow.com/a/49708522/2578823) about this, hope this may help others.
